pkg/apiserver/resthandler.go

```diff
@@ -759,7 +759,7 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 		ctx = api.WithNamespace(ctx, namespace)
 
 		options := &api.DeleteOptions{}
-		if checkBody {
+		if isGracefulDeletion {
```

Comment:
Probably better for it to be `allowsOptions` (now that you support query parameters).  We may have sub resources that support delete that have different api objects instead of DeleteOptions in the future and this would be one step towards that.